app/models/manageiq/providers/amazon/agent_coordinator_worker/runner/response_thread.rb

```diff
@@ -0,0 +1,190 @@
+  require 'scanning_mixin'
+  include ScanningMixin
+
+  #
+  # This thread startup method is called by the AgentCoordinatorWorker::Runner
+  # at the end of the do_before_work_loop method.
+  #
+  def start_response_thread
+    Thread.new do
+      _log.info("Starting AWS SSA Response Thread")
+      begin
+        # Wait a bit for the Agent Coordinators to be started
+        sleep(20)
+        wait_for_responses
+      rescue SystemExit
+        _log.info("SystemExit received, exiting AWS SSA Response Thread")
+        stop_response_thread
+        Thread.exit
+      rescue => err
+        _log.error("start_response_thread: Error [#{err}]")
+        _log.error("start_response_thread: Error - [#{err.backtrace.join("\n")}]")
+      end
+    end
+  end
+
+  def stop_response_thread
+    @shutdown_instance_wait_thread = true
+  end
+
+  def wait_for_responses
+    @shutdown_instance_wait_thread = nil
+    until @shutdown_instance_wait_thread
+      @coordinators.each do |coord|
+        _log.debug("Checking replies for Agent Coordinator Provider #{coord.ems.name}")
+        break if @shutdown_instance_wait_thread
+        if coord.reply_queue_empty?
+          _log.debug("No Replies visible for Provider #{coord.ems.name}")
+        else
+          ssaq_args = {}
+          ssaq_args[:ssa_bucket]    = coord.ssa_bucket
+          ssaq_args[:reply_queue]   = coord.reply_queue
+          ssaq_args[:request_queue] = coord.request_queue
+          ssaq_args[:region]        = coord.ems.provider_region
+          ssaq_args[:sqs]           = coord.sqs
+          ssaq_args[:s3]            = coord.s3
+          ssaq                      = AmazonSsaSupport::SsaQueue.new(ssaq_args)
+          unless ssaq
+            _log.error("Error creating SsaQueue for #{coord.ems.name}")
+            next
+          end
+          _log.debug("Getting replies for #{coord.ems.name}")
+          begin
+            ssaq.reply_loop do |reply|
+              _log.debug("Reply for #{coord.ems.name}: #{reply}")
+              perform_metadata_sync(reply)
+            end
+          rescue => err
+            _log.error("Error #{err} processing replies for #{coord.ems.name}.  Continuing.")
+            next
+          end
+        end
+      end
+      break if @shutdown_instance_wait_thread
+      # While this is WIP we will hard code this to 10 seconds.
+      # reponse_check_sleep_seconds = response_thread_sleep_seconds
+      response_check_sleep_seconds = 10
+      _log.debug("going to sleep for #{response_check_sleep_seconds} seconds after getting all Agent Coordinators")
+      sleep(response_check_sleep_seconds) unless @shutdown_instance_wait_thread
+    end
+  end
+
+  def perform_metadata_sync(extract_reply)
+    ost       = OpenStruct.new
+    ost.reply = extract_reply
+    ost.jobid = extract_reply[:job_id]
+    job       = Job.find_by(:id => ost.jobid)
+    raise _("Unable to sync data for job with id <%{number}>. Job not found.") % {:number => ost.jobid} if job.nil?
```

Comment:
Why not just use `#{ost.jobid}`?

Reply:
Some of this code was "acquired" from the VmScan and ScanningMixins code.  This line did look overly complicated and I will change it.